nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -790,20 +796,108 @@ contract StakingEscrow is Issuer, IERC900History {
         uint256 requestedLockedTokens = _value.add(lockedTokens);
         require(requestedLockedTokens <= info.value && requestedLockedTokens <= maxAllowableLockedTokens);
 
-        uint16 duration = _periods;
         // next period is committed
         if (info.nextCommittedPeriod == nextPeriod) {
-            // if winding down is enabled and next period is committed
-            // then sub-stakes duration were decreased
-            if (info.flags.bitSet(WIND_DOWN_INDEX)) {
-                duration -= 1;
-            }
             lockedPerPeriod[nextPeriod] += _value;
             emit CommitmentMade(_staker, nextPeriod, _value);
         }
-        saveSubStake(info, nextPeriod, 0, duration, _value);
 
-        emit Locked(_staker, _value, nextPeriod, _periods);
+        // if index was provided then increase existing sub-stake
+        if (_index < MAX_SUB_STAKES) {
```

Comment:
Idea is if provided index is not existing (> max length) then create new one, and I don't really want to create new constant just for this internal twist

Reply:
I'm for documenting this because it's fully internal thing, also public constant is costly because compiler will generate getter (internal constant is not costly)